Plugins/StandardLibraryBuilderPlugin/StandardLibraryBuilderPlugin.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import PackagePlugin
+
+/// The Swift Package Manager plugin that builds the standard library.
+@main
+struct StandardLibraryBuilderPlugin: SPMBuildToolPlugin {
+
+  func buildCommands(
+    context: PackagePlugin.PluginContext, target: PackagePlugin.Target
+  ) throws -> [SPMBuildCommand] {
+    let sourceDirectory = URL(fileURLWithPath: #filePath)
+      .deletingLastPathComponent().deletingLastPathComponent().deletingLastPathComponent()
+      / "StandardLibrary" / "Sources"
+
+    print("SourceDirectory: ", sourceDirectory)
+    let hostedSources = FileManager.default.enumerator(
+      at: sourceDirectory,
+      includingPropertiesForKeys: [.isRegularFileKey],
+      options: [.skipsHiddenFiles, .skipsPackageDescendants])!
+      .compactMap { $0 as? URL }
+      .filter { $0.pathExtension == "hylo" }
+
+    let freestrandingDirectory = sourceDirectory / "Core"
+    let freestandingSources = hostedSources.filter(freestrandingDirectory.contains)
```

Comment:
Thar's a algorithm for that.

```suggestion
    let freestandingSources = hostedSources.filter { $0.pathComponents.starts(with: freestrandingDirectory.pathComponents) }
```

I didn't put it in `contains` because that function doesn't do what it says on the tin if you read the comment carefully.

Reply:
Sorry, which function doesn't do what it says it does, and on what?

Reply:
`URL.contains`, whose documentation says “returns `true` if `self` is the path of a directory containing `other`.”.  In fact it doesn't tell you anything whether `self` refers to a directory (which requires verifying that it's a `file://` URL and then looking at disk) or whether it contains `other`. in fact it only tells you whether the path of `self` is a parent of the path of `other`.  Potential for little inaccuracies (or at the very least, ambiguities) like that one,  is among the many problems with  `URL` that make me want to write a first-class path abstraction.